src/PluginManager.js

```diff
@@ -150,8 +106,75 @@ class PluginManager {
         this.add(new BuiltinPlugin({
             getLogger: log4js.getLogger.bind(log4js)
         }));
+
+        this.resolve = createRequire(import.meta.url).resolve;
     }
 
+    /*
+     * Attempt to load the plugin in various places:
+     *
+     * - from the node_modules where ilib-lint was loaded
+     * - from the current directory's node_modules
+     * - from the plugins directory one directory up
+     *
+     * Each time it attempts to load it, it will try two ways:
+     *
+     * - As-is. Maybe it is a fully specified package name?
+     * - With the "ilib-lint-" prefix. Try again except with
+     * the prefix. This allows the users to configure plugins
+     * in the config file more tersely, similar to the way
+     * babel plugins can be named with only the unique part.
+     * The old "i18nlint-" prefix is no longer accepted in
+     * either the name of the directory or the name of the
+     * package.
+     *
+     * @private
+     */
+    loadPlugin(name) {
+        let pkgName = name;
+        return attemptLoad(pkgName).catch(e1 => {
+            logger.trace(e1);
+            pkgName = `ilib-lint-${name}`;
+            return attemptLoad(pkgName).catch(e2 => {
+                logger.trace(e2);
+                pkgName = path.join(process.cwd(), "node_modules", name);
+                return attemptLoadPath(pkgName).catch(e3 => {
+                    logger.trace(e3);
+                    pkgName = path.join(process.cwd(), "node_modules", `ilib-lint-${name}`);
+                    return attemptLoadPath(pkgName).catch(e4 => {
+                        logger.trace(e4);
+                        pkgName = path.join(process.cwd(), "..", "plugins", name + ".js")
+                        return attemptLoad(pkgName).catch(e5 => {
+                            logger.trace(e5);
+                            // on the last attempt, don't catch the exception. Just let it
+                            // go to the overall `catch` clause below.
+                            pkgName = path.join(process.cwd(), "..", "plugins", `ilib-lint-${name}` + ".js")
+                            return attemptLoad(pkgName);
+                        });
+                    });
+                });
+            });
```

Comment:
Nit: you could change this function to async and convert callbacks to a loop that IMO is easier to read e.g.
```js
        const maybeNames = [
            name,
            `ilib-lint-${name}`,
            path.join(process.cwd(), "node_modules", name),
            path.join(process.cwd(), "node_modules", `ilib-lint-${name}`),
            path.join(process.cwd(), "..", "plugins", name + ".js"),
            path.join(process.cwd(), "..", "plugins", `ilib-lint-${name}` + ".js"),
        ];

        let pkgName, module;
        while ((pkgName = maybeNames.shift())) {
            try {
                module = await attemptLoad(pkgName);
                break;
            } catch (e) {
                logger.trace(e);
            }
        }
        if (!module) {
            throw new Error(`Failed to load plugin from any of the expected paths`);
        }
```

Reply:
We should do that in a separate "clean up" PR.